membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -103,6 +100,73 @@ class AccountController(commonActions: CommonActions, override val controllerCom
     }
   }
 
+  private def updateDD[P <: SubscriptionPlan.AnyPlan : SubPlanReads] = BackendFromCookieAction.async { implicit request =>
+    // TODO - refactor to use the Zuora-only based lookup, like in AttributeController.pickAttributes - https://trello.com/c/RlESb8jG
+
+    val updateForm = Form { tuple(
+      "accountName" -> nonEmptyText,
+      "accountNumber" -> nonEmptyText,
+      "sortCode" -> nonEmptyText
+    ) }
+
+    val tp = request.touchpoint
+    val maybeUserId = authenticationService.userId
+    logger.info(s"Attempting to update direct debit for $maybeUserId")
+    (for {
+      user <- EitherT(Future.successful(maybeUserId \/> "no identity cookie for user"))
+      directDebitDetails <- EitherT(Future.successful(updateForm.bindFromRequest().value \/> "no direct debit details submitted with request"))
+      (bankAccountName, bankAccountNumber, bankSortCode) = directDebitDetails
+      sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"no SF user $user")))
+      subscription <- EitherT(tp.subService.current[P](sfUser).map(_.headOption).map (_ \/> s"no current subscriptions for the sfUser $sfUser"))
+      account <- EitherT(annotateFailableFuture(tp.zuoraService.getAccount(subscription.accountId), s"get account with id ${subscription.accountId}"))
```

Comment:
I think it might be possible to get all of the data we need here (about the account and billing contact) from a single call: https://www.zuora.com/developer/api-reference/#operation/GET_AccountSummary

Reply:
Ah, in fact I think you might already be using it under the hood: https://github.com/guardian/membership-common/blob/401f38f5c5de418429b03e0e94982e17a547db2a/src/main/scala/com/gu/zuora/rest/ZuoraRestService.scala#L309-L311

Reply:
So hopefully we don't need the call to `tp.zuoraService.getContact`?

Reply:
unfortunately the current case class for `BillToContact` that forms part of the `AccountSummary` which one gets back from the Zuora REST service, doesn't include the fields I need (first name, last name etc). It might well be the case that just need to update https://github.com/guardian/membership-common/blob/401f38f5c5de418429b03e0e94982e17a547db2a/src/main/scala/com/gu/zuora/rest/ZuoraRestService.scala#L191-L194 to add those fields as they might be present in the raw response from Zuora - but that would involve a change to https://github.com/guardian/membership-common which I've managed to avoid thus far.

Reply:
If that's all we need to do to save the call then I would go for it now. It's only a small membership-common change which shouldn't affect any other apps, so it should be relatively straightforward.

Reply:
On closer inspection, the `CreatePaymentMethod` further down the for comprehension requires a Contact of the soap variety, so I would need to rebuild the 6fields case class in order to reuse the current DD implementation from membership-common which re-raises the question of doing the DD from scratch with rest as new functionality in membership-common.